--- hake/Config.hs.template	2016-09-15 16:38:07.624250233 +0200
+++ hake/Config.hs.template.cab	2016-09-15 16:48:36.483738250 +0200
@@ -33,7 +33,7 @@
 k1om_toolspec    :: Maybe (Maybe FilePath -> ToolDetails)
 
 -- Default toolchains
-arm_tools     = fromMaybe Tools.arm_system
+arm_tools     = fromMaybe Tools.arm_ethz_rhel7
                           arm_toolspec
                 toolroot
 aarch64_tools = fromMaybe Tools.arm_netos_linaro_aarch64_2014_11
@@ -132,15 +132,15 @@
 -- manipulate ELF files (arm_boot and usbboot).
 -- (defaults for Ubuntu w/ libelf-freebsd and freebsd-glue)
 libelf_include :: String
-libelf_include = "/usr/include/freebsd"
+libelf_include = "/pub/aos/tools/libelf/usr/include"
 libelf_link :: String
-libelf_link = "-lelf-freebsd"
+libelf_link = "/pub/aos/tools/libelf/usr/lib/libelf.a"
 
 -- Libusb include path and linker options for building usbboot
 libusb_include :: String
-libusb_include = "/usr/include/libusb-1.0"
+libusb_include = "/pub/aos/tools/libusb-1.0/include/libusb-1.0"
 libusb_libdir :: String
-libusb_libdir = "/usr/lib"
+libusb_libdir = "/pub/aos/tools/libusb-1.0/lib"
 
 -- Optimisation flags (-Ox -g etc.) passed to compiler
 cOptFlags :: [String]
