From 4a7a5273b1dd5eb762df09ad60fbd5186572f942 Mon Sep 17 00:00:00 2001
From: Simon Gerber <simon.gerber@inf.ethz.ch>
Date: Wed, 5 Oct 2016 08:36:11 +0200
Subject: [PATCH] remove implementation details from include/mm/mm.h

---
 include/mm/mm.h | 19 +++++--------------
 1 file changed, 5 insertions(+), 14 deletions(-)

diff --git a/include/mm/mm.h b/include/mm/mm.h
index 930ef55..93f17e1 100644
--- a/include/mm/mm.h
+++ b/include/mm/mm.h
@@ -29,22 +29,12 @@ enum nodetype {
     NodeType_Allocated  ///< This region exists and is allocated
 };
 
-struct capinfo {
-    struct capref cap;
-    genpaddr_t base;
-    gensize_t size;
-};
-
 /**
- * \brief Node in Memory manager
+ * An element of the data structure you will use to keep track of regions in a
+ * memory manager.
  */
 struct mmnode {
-    enum nodetype type;    ///< Type of `this` node.
-    struct capinfo cap;    ///< Cap in which this region exists
-    struct mmnode *prev;   ///< Previous node in the list.
-    struct mmnode *next;   ///< Next node in the list.
-    genpaddr_t base;       ///< Base address of this region
-    gensize_t size;        ///< Size of this free region in cap
+    // TODO: define fields for your data structure
 };
 
 /**
@@ -59,7 +49,8 @@ struct mm {
     slot_refill_t slot_refill;   ///< Slot allocator refill function
     void *slot_alloc_inst;       ///< Opaque instance pointer for slot allocator
     enum objtype objtype;        ///< Type of capabilities stored
-    struct mmnode *head;         ///< Head of doubly-linked list of nodes in order
+    struct mmnode *head;         ///< Head of data structure keeping track of
+                                 ///  individual regions managed by this allocator
 };
 
 errval_t mm_init(struct mm *mm, enum objtype objtype,
-- 
2.7.4

